datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -115,34 +114,6 @@ impl ColumnBounds {
     }
 }
 
-/// Represents the bounds for all join key columns from a single partition.
-/// This contains the min/max values computed from one partition's build-side data.
-#[derive(Debug, Clone)]
-struct PartitionBounds {
```

Comment:
I think keeping the struct is helpful from a readability perspective to avoid nested vecs

Reply:
Yes please!